src/config/scrivitoSites.ts

```diff
@@ -37,8 +37,55 @@ export function siteForUrl(
 
 export async function ensureSiteIsPresent() {
   if ((await load(currentSiteId)) === null) {
-    navigateTo(() =>
-      Obj.onAllSites().where('_path', 'equals', '/').order('_language').first(),
-    )
+    navigateTo(() => {
+      const sites = Obj.onAllSites().where('_path', 'equals', '/').toArray()
+      const preferredLanguageOrder = [...window.navigator.languages, 'en', null]
+
+      for (const language of preferredLanguageOrder) {
+        const site = sites.find((site) => siteHasLanguage(site, language))
+        if (site) return site
+      }
+
+      return sites[0] || null
+    })
+  }
+}
+
+function siteHasLanguage(site: Obj, language: string | null) {
+  const siteLanguage = site.language()
+  return language && siteLanguage
+    ? language.startsWith(siteLanguage)
+    : language === siteLanguage
+}
+
+export function getTenantFromEnv(): string | undefined {
+  if (import.meta.env.SCRIVITO_TENANT) return import.meta.env.SCRIVITO_TENANT
+
+  if (!location) throw new Error('Could not determine tenant!')
+
+  // Multitenancy mode
+  const tenantFromUrl = location.pathname.match(/^\/([0-9a-f]{32})\b/)?.[1]
+  const tenantFromQuery = new URLSearchParams(location.search).get('tenantId')
+  const tenant = tenantFromUrl || tenantFromQuery
+
+  if (!tenant) {
+    if (
+      import.meta.env.VITE_MULTITENANCY_FALLBACK_SCRIVITO_TENANT &&
+      !tenantFromQuery
+    ) {
+      const fallbackScrivitoTenant = import.meta.env
+        .VITE_MULTITENANCY_FALLBACK_SCRIVITO_TENANT
+      if (
+        typeof fallbackScrivitoTenant === 'string' &&
+        fallbackScrivitoTenant.match(/^[0-9a-f]{32}$/)
+      ) {
+        location.replace(`${location.origin}/${fallbackScrivitoTenant}`)
+        return
+      }
+    }
+
+    throw new Error('Could not determine tenant!')
   }
+
+  return tenant
 }
```

Comment:
Where does this come from? Is this still needed?

Reply:
No! Somehow the main merge failed to properly remove it.